pkg/agent/util/iptables/iptables.go

```diff
@@ -184,47 +185,60 @@ func matchProtocol(ipt *iptables.IPTables, protocol Protocol) bool {
 	return false
 }
 
-// DeleteRule checks if target rule already exists, deletes the rule if found.
-func (c *Client) DeleteRule(table string, chain string, ruleSpec []string) error {
-	for idx := range c.ipts {
-		exist, err := c.ipts[idx].Exists(table, chain, ruleSpec...)
+// DeleteRules checks if target rule already exists in all iptables, deletes the rule if found.
+func (c *Client) DeleteRules(protocol Protocol, table string, chain string, ruleSpec []string) error {
```

Comment:
`DeleteRule` to align with `InsertRule` and `AppendRule`

Reply:
Updated.